### PR TITLE
Strategy Deals Functionality

### PR DESCRIPTION
**Pull request by jitendra-xoriant**
Added Deals Functionality to Strategy.
GET and POST support to Deals Entity with Media API Base

### CODE REVIEW

File: src/main/java/com/mediamath/terminalone/models/CreativeApproval.java [hunk 1]
```diff
@@ -1,170 +1,178 @@
...
-    return null;
-  }
-
-  @Override
-  public String getUri() {
-    return null;
-  }
-
-}
+/*******************************************************************************
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/CreativeApproval.java]: Is this a new file? Why so much changes?

File: src/main/java/com/mediamath/terminalone/models/Strategy.java [hunk 2]
```diff
@@ -1,679 +1,697 @@
...
-	public Form getForm() {
-		return null;
-	}
-
-	@Override
-	public String getUri() {
-		return null;
-	}
-
-}
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/Strategy.java]: If entire file is shown in diff, it is very difficult to review. Can you tell me which part is the actual change?

File: src/main/java/com/mediamath/terminalone/service/PostService.java [hunk 3]
```diff
@@ -238,7 +243,6 @@ public Strategy save(Strategy entity) throws ClientException, ParseException {
 				strategy = (Strategy) finalJsonResponse.getData();
 			}
 		}
-
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/service/PostService.java]: extra space

File: src/main/java/com/mediamath/terminalone/models/CreativeApproval.java (hunk 1 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/CreativeApproval.java]: Need to check the reason why this is happening.

File: src/main/java/com/mediamath/terminalone/models/Strategy.java (hunk 2 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/Strategy.java]: Added 2 lists regarding deals

File: src/main/java/com/mediamath/terminalone/service/PostService.java (hunk 3 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/service/PostService.java]: ok

File: src/main/java/com/mediamath/terminalone/models/Deal.java [hunk 4]
```diff
@@ -51,7 +51,7 @@
   private mediaTypes media_type;
   private String name;
   private boolean partner_sourced;
-  private float price;
+  private Currency price;
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/Deal.java]: Why is this change introduced? Is Currency an encapsulation for price and the currency?

File: src/main/java/com/mediamath/terminalone/models/Owner.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.mediamath.terminalone.models;
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/Owner.java]: This new model is to account for the deals functionality?

File: src/main/java/com/mediamath/terminalone/models/Permissions.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
+package com.mediamath.terminalone.models;
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/Permissions.java]: New model? Where all its used?

File: src/main/java/com/mediamath/terminalone/models/Deal.java (hunk 4 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/Deal.java]: yes. Added as per documentation.

File: src/main/java/com/mediamath/terminalone/models/Owner.java (hunk 5 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/Owner.java]: only for deal model.

File: src/main/java/com/mediamath/terminalone/models/Permissions.java (hunk 6 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/Permissions.java]: only for deal model

**Pull request merged**